lib/cadet/assessments/assessments.ex

```diff
@@ -753,6 +809,69 @@ defmodule Cadet.Assessments do
     |> Repo.transaction()
   end
 
+  def update_contest_voting_submission_status(submission = %Submission{}, question = %Question{}) do
+    count_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> Repo.all()
+      )
+
+    count_not_nil_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> where([c], not is_nil(c.score))
+        |> Repo.all()
+      )
+
+    if count_entries == count_not_nil_entries do
+      submission |> Submission.changeset(%{status: :attempted}) |> Repo.update()
+    end
```

Comment:
Why not just check that is_nil entries are zero?

You can use `Repo.exists?` in that case.